src/api/spec/controllers/source_controller_spec.rb

```diff
@@ -50,4 +51,18 @@
     it { expect(response).to be_success }
     it { expect(project.config.to_s).to include('Updated', 'by', 'test') }
   end
+
+  describe "PUT #update_project_meta" do
+    it "raises exception for invalid DownloadRepository" do
+      login admin
+      bypass_rescue
+
+      xml = ActiveXML::Node.new(project.to_axml)
+      repo = xml.add_element 'repository', { name: "download_on_demand" }
+      repo.add_element 'download', { arch: "i586", url: "http://somewhere", repotype: "INVALID" }
+      request.env['RAW_POST_DATA'] = xml.dump_xml
+      expect { post :update_project_meta, params: { project: project } }.to raise_error(Project::SaveError, /Repotype is not included in the list/)
```

Comment:
hmmm, yes thats annoying to test :/

ok fair enough then using `bypass_rescue` :+1: